To: vim_dev@googlegroups.com
Subject: Patch 7.3.457
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.457
Problem:    When setting $VIMRUNTIME later the directory for fetching
            translated messages is not adjusted.
Solution:   Put bindtextdomain() in vim_setenv().
Files:      src/misc1.c


*** ../vim-7.3.456/src/misc1.c	2012-02-05 22:05:44.000000000 +0100
--- src/misc1.c	2012-02-26 13:27:28.000000000 +0100
***************
*** 4133,4149 ****
  	{
  	    vim_setenv((char_u *)"VIMRUNTIME", p);
  	    didset_vimruntime = TRUE;
- #ifdef FEAT_GETTEXT
- 	    {
- 		char_u	*buf = concat_str(p, (char_u *)"/lang");
- 
- 		if (buf != NULL)
- 		{
- 		    bindtextdomain(VIMPACKAGE, (char *)buf);
- 		    vim_free(buf);
- 		}
- 	    }
- #endif
  	}
  	else
  	{
--- 4133,4138 ----
***************
*** 4221,4226 ****
--- 4210,4231 ----
  	putenv((char *)envbuf);
      }
  #endif
+ #ifdef FEAT_GETTEXT
+     /*
+      * When setting $VIMRUNTIME adjust the directory to find message
+      * translations to $VIMRUNTIME/lang.
+      */
+     if (*val != NUL && STRICMP(name, "VIMRUNTIME") == 0)
+     {
+ 	char_u	*buf = concat_str(val, (char_u *)"/lang");
+ 
+ 	if (buf != NULL)
+ 	{
+ 	    bindtextdomain(VIMPACKAGE, (char *)buf);
+ 	    vim_free(buf);
+ 	}
+     }
+ #endif
  }
  
  #if defined(FEAT_CMDL_COMPL) || defined(PROTO)
*** ../vim-7.3.456/src/version.c	2012-02-22 19:13:00.000000000 +0100
--- src/version.c	2012-02-29 13:47:09.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     457,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
118. You are on a first-name basis with your ISP's staff.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
